From 049e7f984603fd9261cbe0c594ac8de52b475477 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ville=20Syrj=C3=A4l=C3=A4?= <ville.syrjala@linux.intel.com>
Date: Wed, 3 Jul 2019 23:08:20 +0300
Subject: [PATCH] FROMGIT: drm/i915: Disable sprite gamma on ivb-bdw
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

We don't currently have any use for the sprite gamma on ivb-bdw.
Let's disable it. We already do that on skl+.

On pre-ivb there is no way to disable the sprite gamma, and it
only affects YCbCr pixel formats, whereas on ivb+ it also
affects RGB formats.

Reviewed-by: Maarten Lankhorst <maarten.lankhorst@linux.intel.com>
Signed-off-by: Ville Syrjälä <ville.syrjala@linux.intel.com>
Link: https://patchwork.freedesktop.org/patch/msgid/20190703200824.5971-3-ville.syrjala@linux.intel.com
(cherry picked from commit 423ee8e99aa5ee7ee5133616366099b8379870dc
https://anongit.freedesktop.org/git/drm-intel.git drm-intel-next)
Signed-off-by: Chintan Patel <chintan.m.patel@intel.com>

Conflicts:
	drivers/gpu/drm/i915/intel_sprite.c
(cmpatel): Lot of restructuring happend in drm-tip which has not been brought with this patch.
           Modifying existing file without pulling restructuring.

BUG=chromium:973690
TEST=On BSW board, open Play Store to install any app;
     No screen jitters/black screen should be observed

Change-Id: Ic77b0d96bbdc522f02b574bf43416d86bd6de134
Reviewed-on: https://chromium-review.googlesource.com/c/chromiumos/third_party/kernel/+/1696194
Reviewed-by: Stéphane Marchesin <marcheu@chromium.org>
Tested-by: Chintan M Patel <chintan.m.patel@intel.com>
Commit-Queue: Chintan M Patel <chintan.m.patel@intel.com>

[rebase52(rrangel): Cherry picked original commit423ee8e99aa5ee7ee5133616366099b8379870dc]
Signed-off-by: Raul E Rangel <rrangel@chromium.org>
---
 drivers/gpu/drm/i915/display/intel_sprite.c | 2 ++
 drivers/gpu/drm/i915/i915_reg.h             | 2 +-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/display/intel_sprite.c b/drivers/gpu/drm/i915/display/intel_sprite.c
index 004b52027ae8..c7314617424a 100644
--- a/drivers/gpu/drm/i915/display/intel_sprite.c
+++ b/drivers/gpu/drm/i915/display/intel_sprite.c
@@ -1013,6 +1013,8 @@ static u32 ivb_sprite_ctl(const struct intel_crtc_state *crtc_state,
 		return 0;
 	}
 
+	sprctl |= SPRITE_INT_GAMMA_DISABLE;
+
 	if (plane_state->base.color_encoding == DRM_COLOR_YCBCR_BT709)
 		sprctl |= SPRITE_YUV_TO_RGB_CSC_FORMAT_BT709;
 
diff --git a/drivers/gpu/drm/i915/i915_reg.h b/drivers/gpu/drm/i915/i915_reg.h
index d6483b5dc8e5..2cda055814ca 100644
--- a/drivers/gpu/drm/i915/i915_reg.h
+++ b/drivers/gpu/drm/i915/i915_reg.h
@@ -6457,7 +6457,7 @@ enum {
 #define   SPRITE_YUV_ORDER_VYUY		(3 << 16)
 #define   SPRITE_ROTATE_180		(1 << 15)
 #define   SPRITE_TRICKLE_FEED_DISABLE	(1 << 14)
-#define   SPRITE_INT_GAMMA_ENABLE	(1 << 13)
+#define   SPRITE_INT_GAMMA_DISABLE	(1 << 13)
 #define   SPRITE_TILED			(1 << 10)
 #define   SPRITE_DEST_KEY		(1 << 2)
 #define _SPRA_LINOFF		0x70284
-- 
2.23.0.rc1.153.gdeed80330f-goog

